homeassistant/components/lidarr/__init__.py

```diff
@@ -0,0 +1,85 @@
+"""The Lidarr component."""
+from __future__ import annotations
+
+from aiopyarr.lidarr_client import LidarrClient
+from aiopyarr.models.host_configuration import PyArrHostConfiguration
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_URL, CONF_VERIFY_SSL, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.device_registry import DeviceEntryType
+from homeassistant.helpers.entity import DeviceInfo, EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DEFAULT_NAME, DOMAIN
+from .coordinator import (
+    DiskSpaceDataUpdateCoordinator,
+    LidarrDataUpdateCoordinator,
+    QueueDataUpdateCoordinator,
+    StatusDataUpdateCoordinator,
+    WantedDataUpdateCoordinator,
+)
+
+PLATFORMS = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Lidarr from a config entry."""
+    host_configuration = PyArrHostConfiguration(
+        api_token=entry.data[CONF_API_KEY],
+        verify_ssl=entry.data[CONF_VERIFY_SSL],
+        url=entry.data[CONF_URL],
+    )
+    lidarr = LidarrClient(
+        host_configuration=host_configuration,
+        session=async_get_clientsession(hass, host_configuration.verify_ssl),
+        request_timeout=60,
+    )
+    coordinators: dict[str, LidarrDataUpdateCoordinator] = {
+        "disk_space": DiskSpaceDataUpdateCoordinator(hass, host_configuration, lidarr),
+        "queue": QueueDataUpdateCoordinator(hass, host_configuration, lidarr),
+        "status": StatusDataUpdateCoordinator(hass, host_configuration, lidarr),
+        "wanted": WantedDataUpdateCoordinator(hass, host_configuration, lidarr),
+    }
+    # Temporary, until we add diagnostic entities
+    _version = None
+    for coordinator in coordinators.values():
+        await coordinator.async_config_entry_first_refresh()
+        if isinstance(coordinator, StatusDataUpdateCoordinator):
+            _version = coordinator.data
```

Comment:
Since `version` is defined outside of the loop, the `wanted` coordinator will also have this version applied. Is that on purpose?

Reply:
Yes. Software version only comes from `async_get_system_status`. his is just the Lidarr software version.